services-js/registry-certs/client/birth/ReviewRequest.tsx

```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import Head from 'next/head';
+import Router from 'next/router';
+
+import { css } from 'emotion';
+
+import {
+  CHARLES_BLUE,
+  FREEDOM_RED,
+  GRAY_100,
+  OPTIMISTIC_BLUE,
+} from '@cityofboston/react-fleet';
+
+import { PageDependencies } from '../../pages/_app';
+
+import PageLayout from '../PageLayout';
+import CostSummary from '../common/CostSummary';
+
+import { BREADCRUMB_NAV_LINKS } from './constants';
+
+import {
+  BUTTONS_CONTAINER_STYLING,
+  SECONDARY_BUTTON_STYLE,
+} from './questions/styling';
+
+interface Props extends Pick<PageDependencies, 'birthCertificateRequest'> {}
+
+interface State {
+  certificateQuantity: number;
+}
+
+/**
+ * Component which allows a user to review their request, and update the
+ * quantity of birth certificates they are requesting.
+ *
+ * User can proceed to /checkout, go back to the questions flow, or
+ * clear all information and start over.
+ */
+export default class ReviewRequest extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      certificateQuantity: 1,
+    };
+  }
+
+  handleQuantityChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // Update quantity; if user erases value in field, quantity will return to 1
+    this.setState({ certificateQuantity: +event.target.value || 1 });
+  };
+
+  // todo: warn user before resetting information (i.e. “are you sure? y/n” dialog)
+  userResetStartOver = () => {
+    this.props.birthCertificateRequest.clearBirthCertificateRequest();
+
+    this.returnToQuestions();
+  };
+
+  returnToQuestions = () => {
+    // There is a check in componentDidMount to see if the required field
+    // for the last question has been answered; this will bring the user
+    // back to that last question.
+    Router.push('/birth');
+  };
+
+  goToCheckout = () => {
+    this.props.birthCertificateRequest.setQuantity(
```

Comment:
Do we want state as an intermediary or should the quantity come from `birthCertificateRequest` directly?

I'm thinking of the case of updating the quantity, then going back to questions, then coming forward to the review and the quantity being different.

Reply:
👍 I guess we might as well drop the component’s own state in favor of `birthCertificateRequest` managing it fully.

Reply:
Works for me.